scipy/spatial/tests/test_distance.py

```diff
@@ -982,12 +1366,18 @@ def _check_calling_conventions(self, X, metric, eps=1e-07, **kwargs):
             if verbose > 2:
                 print(e_cls.__name__)
                 print(e)
-            assert_raises(e_cls, pdist, X, metric=metric, **kwargs)
-            assert_raises(e_cls, pdist, X, metric=eval(metric), **kwargs)
-            assert_raises(e_cls, pdist, X, metric="test_" + metric, **kwargs)
+            try:
+                assert_raises(Exception, pdist, X, metric=metric, **kwargs)
+                assert_raises(Exception, pdist, X, metric=eval(metric), **kwargs)
+                assert_raises(Exception, pdist, X, metric="test_" + metric, **kwargs)
+            except Exception as e2:
```

Comment:
I don't think it makes sense to expect exceptions coming from C to be of exactly the same type as those coming from Python--really what is the difference between "TypeError" and "ValueError" for data types anyway? There were a few functions where this was causing an issue, iirc